src/citrine/informatics/design_spaces/product_design_space.py

```diff
@@ -11,56 +12,75 @@
-    status = properties.String('status', serializable=False)
-    status_info = properties.Optional(
-        properties.List(properties.String()),
-        'status_info',
-        serializable=False
-    )
-    archived = properties.Boolean('archived', default=False)
-    experimental = properties.Boolean("experimental", serializable=False, default=True)
-    experimental_reasons = properties.Optional(
-        properties.List(properties.String()),
-        'experimental_reasons',
-        serializable=False
+    subspaces = properties.List(properties.Union(
+        [properties.UUID, properties.Object(DesignSpace)]
+    ), 'config.subspaces', default=[])
+    dimensions = properties.Optional(
+        properties.List(properties.Object(Dimension)), 'config.dimensions'
     )
+    typ = properties.String('config.type', default='ProductDesignSpace', deserializable=False)
+    # Product design spaces may not be embedded in other subspaces, hence status is required
+    status = properties.String('status', serializable=False)
 
     # NOTE: These could go here or in _post_dump - it's unclear which is better right now
     module_type = properties.String('module_type', default='DESIGN_SPACE')
     schema_id = properties.UUID('schema_id', default=UUID('6c16d694-d015-42a7-b462-8ef299473c9a'))
 
-    def __init__(self,
+    def __init__(self, *,
                  name: str,
                  description: str,
-                 dimensions: List[Dimension],
+                 subspaces: Optional[List[Union[UUID, DesignSpace]]] = None,
+                 dimensions: Optional[List[Dimension]] = None,
                  session: Session = Session()):
         self.name: str = name
         self.description: str = description
-        self.dimensions: List[Dimension] = dimensions
+        self.subspaces: List[Union[UUID, DesignSpace]] = subspaces or []
+        self.dimensions: List[Dimension] = dimensions or []
         self.session: Session = session
 
+    def dump(self) -> dict:
+        """Override dump to replace on-platform subspaces with their uids."""
+        model_copy = deepcopy(self)
+        for i, subspace in enumerate(model_copy.subspaces):
+            if subspace.uid is not None:
+                model_copy.subspaces[i] = subspace.uid
+        serialized = properties.Object(ProductDesignSpace).serialize(model_copy)
+        return self._post_dump(serialized)
+
     def _post_dump(self, data: dict) -> dict:
         data['display_name'] = data['config']['name']
+        for i, subspace in enumerate(data['config']['subspaces']):
+            if isinstance(subspace, dict):
+                # embedded design spaces are not modules, so only serialize their config
+                data['config']['subspaces'][i] = subspace['config']
+        return data
+
+    @classmethod
+    def _pre_build(cls, data: dict) -> dict:
+        subspaces = data['config'].get('subspaces', [])
+        for i, _ in enumerate(subspaces):
```

Comment:
This is just propagating the instance -> config rename deeper into the tree, but not changing the shape, right?  If so, that's probably worth a comment.